servers/src/main/java/tachyon/worker/block/evictor/EvictorUtils.java

```diff
@@ -16,72 +16,92 @@
     // reassure the plan is feasible: enough free space to satisfy bytesToBeAvailable, and enough
     // space in lower tier to move blocks in upper tier there
-    Map<Integer, Long> bytesToBeAvailableInTier =
-        new HashMap<Integer, Long>(metaManager.getTiers().size());
-    List<Integer> tierAliases = new ArrayList<Integer>();
 
-    List<Long> blockIds = new ArrayList<Long>(plan.toEvict().size() + plan.toMove().size());
-    blockIds.addAll(plan.toEvict());
-    for (Pair<Long, BlockStoreLocation> move : plan.toMove()) {
-      blockIds.add(move.getFirst());
+    // Map from dir to a pair of bytes to be available in this dir and bytes to move into this dir
+    // after the plan taking action
+    Map<StorageDir, Pair<Long, Long>> spaceInfoInDir = new HashMap<StorageDir, Pair<Long, Long>>();
+
+    for (long blockId : plan.toEvict()) {
+      BlockMeta block = metaManager.getBlockMeta(blockId);
+      StorageDir dir = block.getParentDir();
+      if (spaceInfoInDir.containsKey(dir)) {
+        Pair<Long, Long> spaceInfo = spaceInfoInDir.get(dir);
+        spaceInfo.setFirst(spaceInfo.getFirst() + block.getBlockSize());
+      } else {
+        spaceInfoInDir.put(dir, new Pair<Long, Long>(
+            dir.getAvailableBytes() + block.getBlockSize(), 0L));
+      }
     }
 
-    for (long blockId : blockIds) {
+    for (Pair<Long, BlockStoreLocation> move : plan.toMove()) {
+      long blockId = move.getFirst();
       BlockMeta block = metaManager.getBlockMeta(blockId);
-      BlockStoreLocation blockDir = block.getBlockLocation();
       long blockSize = block.getBlockSize();
-      int tierAlias = blockDir.tierAlias();
-      if (bytesToBeAvailableInTier.containsKey(tierAlias)) {
-        bytesToBeAvailableInTier
-            .put(tierAlias, bytesToBeAvailableInTier.get(tierAlias) + blockSize);
+      StorageDir srcDir = block.getParentDir();
+      StorageDir destDir = metaManager.getDir(move.getSecond());
+
+      if (spaceInfoInDir.containsKey(srcDir)) {
+        Pair<Long, Long> spaceInfo = spaceInfoInDir.get(srcDir);
+        spaceInfo.setFirst(spaceInfo.getFirst() + blockSize);
+      } else {
+        spaceInfoInDir
+            .put(srcDir, new Pair<Long, Long>(srcDir.getAvailableBytes() + blockSize, 0L));
+      }
+
+      if (spaceInfoInDir.containsKey(destDir)) {
+        Pair<Long, Long> spaceInfo = spaceInfoInDir.get(destDir);
+        spaceInfo.setSecond(spaceInfo.getSecond() + blockSize);
       } else {
-        tierAliases.add(tierAlias);
-        bytesToBeAvailableInTier
-            .put(tierAlias, metaManager.getAvailableBytes(blockDir) + blockSize);
+        spaceInfoInDir.put(destDir, new Pair<Long, Long>(destDir.getAvailableBytes(), blockSize));
+      }
+    }
+
+    int firstTierAlias = Integer.MAX_VALUE;
+    for (StorageDir dir : spaceInfoInDir.keySet()) {
+      firstTierAlias = Math.min(firstTierAlias, dir.getParentTier().getTierAlias());
+    }
+    long firstTierDirMaxSpace = Long.MIN_VALUE;
```

Comment:
variable name `firstTierDirMaxSpace` is very confusing. Please either name it more descriptive if possible, or comment this variable.